test/truffle/ecosystem/launchers.sh

```diff
@@ -11,21 +23,58 @@ function check_launchers() {
     [[ "$(${1}gem --version)" =~ ^2.5.2.1$ ]]
     [[ "$(${1}irb --version)" =~ ^irb\ 0.9.6 ]]
     [[ "$(${1}rake --version)" =~ ^rake,\ version\ [0-9.]+ ]]
-    # [[ "$(${1}rdoc --version)" =~ ^4.2.1$ ]] # TODO (pitr-ch 30-Apr-2017): reports 4.3.0 on CI
-    # [[ "$(${1}ri --version)" =~ ^ri\ 4.2.1$ ]] # TODO (pitr-ch 30-Apr-2017): reports 4.3.0 on CI
+    [[ "$(${1}rdoc --version)" =~ ^4.2.1$ ]]
+    [[ "$(${1}ri --version)" =~ ^ri\ 4.2.1$ ]]
 }
 
+function check_in_dir() {
+    cd $1
+    pwd
+    echo "** Check all launchers work in $1 dir"
+    check_launchers "./" true
+    echo "** Check all launchers work in $1 dir using -S option"
+    check_launchers "./truffleruby -S "
+    cd -
+}
+
+
 echo '** Check all launchers work'
 check_launchers bin/ true
+check_in_dir bin
+
+if [[ "$(bin/ruby -Xgraal.warn_unless=false -e "p Truffle.graalvm?")" =~ true ]]
+then
+    check_in_dir ../../bin      # graalvm/jre/bin
+    check_in_dir ../../../bin   # graalvm/bin
+fi
+
+
+echo '** Check gem executables are installed in all bin dirs'
+
+home=$(pwd)
+
+cd "$(dirname $SELF_PATH)/hello-world"
+"$home/bin/gem" build hello-world.gemspec
+"$home/bin/gem" install hello-world-0.0.1.gem
+cd -
 
-cd bin
+version="$(bin/ruby -Xgraal.warn_unless=false -v)"
+test "$(bin/hello-world.rb)" = "Hello world! from $version"
+if [[ "$(bin/ruby -Xgraal.warn_unless=false -e "p Truffle.graalvm?")" =~ true ]]
+then
+    test "$(../../bin/hello-world.rb)" = "Hello world! from $version"
+    test "$(../../../bin/hello-world.rb)" = "Hello world! from $version"
+fi
 
-echo '** Check all launchers work from bin dir'
-check_launchers "./" true
+bin/gem uninstall hello-world -x
 
+test ! -f "bin/hello-world.rb"
+if [[ "$(bin/ruby -Xgraal.warn_unless=false -e "p Truffle.graalvm?")" =~ true ]]
+then
+    test ! -f "../../bin/hello-world.rb"
+    test ! -f "../../../bin/hello-world.rb"
+fi
```

Comment:
This is pretty hard to read due to being written in Bash.
I think it would be nice to rewrite in Ruby in the long term.

Reply:
yeah maybe, that's for another PR though

Reply:
Indeed.